### PR TITLE
[PyGrid] Users CRUD

### PR DESCRIPTION
**Pull request by tcp**
## Description
This fixes the ability to create, read, update and delete users using the PyGrid web API. This PR also introduces the `get_client` and `send_message_with_reply` functions for getting the node client and doing the process of building and sending a message with reply to the client.

Previously, all communication with the node was done directly on the endpoints. That resulted in code being repeated over all endpoints for creating Syft messages and handling the responses. In addition, it required a deeper understanding of how Syft operates in order to properly manage the endpoints. The goal of this abstraction is to reduce complexity by handling Syft messages in a single module in Grid.

The user routes were refactored to use `send_message_with_reply`. A User model was created and should, in the future, be attached to `SyftUser`.

PR Overview:
- `get_client` and `send_message_with_reply` functions
- Adds a User model in Grid
- Handles `get_current_user` errors as 401
- Fixes users crud
- Adds user routes tests
- Cleans up some FastAPI boilerplate code in login

Notes:
- Please merge the [PyGrid tests PR first](https://github.com/OpenMined/PySyft/pull/5882).
- Closes [ch80].

## How has this been tested?
- Added route tests in tests/users
- No tests added to `GridToSyftMessage` though
- There is a lot of room for expanding them

## Checklist
- [x] I have followed the [Contribution Guidelines](https://github.com/OpenMined/.github/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/OpenMined/.github/blob/master/CODE_OF_CONDUCT.md)
- [x] I have commented my code following the [OpenMined Styleguide](https://github.com/OpenMined/.github/blob/master/STYLEGUIDE.md)
- [x] I have labeled this PR with the relevant [Type labels](https://github.com/OpenMined/.github/labels?q=Type%3A)
- [x] My changes are covered by tests

### CODE REVIEW

**Review by iamtrask — APPROVED** [review]
Lovely!

**Comment by tcp** [comment]
@IonesioJunior just merged `dev`. Please review :)

File: packages/grid/backend/app/app/api/deps.py [hunk 1]
```diff
@@ -30,39 +27,24 @@ def get_db() -> Generator:
...
-) -> Any:
+def get_current_user(token: str = Depends(reusable_oauth2)) -> UserPrivate:
     try:
         payload = jwt.decode(
             token, settings.SECRET_KEY, algorithms=[security.ALGORITHM]
         )
         token_data = schemas.TokenPayload(**payload)
-    except (jwt.JWTError, ValidationError):
+        # TODO: Fix jwt.
+        # TODO: Send a secure message with the token instead of fetching the user
```
Comment (madhavajay) [inline packages/grid/backend/app/app/api/deps.py:37]: I guess we need to start storing this somewhere in a tech debt backlog.

**Comment by tcp** [comment]
Closes [ch1121]

**Pull request merged**